test/Atmos/EDMF/closures/surface_functions.jl

```diff
@@ -30,26 +36,71 @@ function subdomain_surface_values(
     atmos::AtmosModel{FT},
     state::Vars,
     aux::Vars,
+    state_int::Vars,
+    aux_int::Vars,
     zLL::FT,
 ) where {FT}
 
     turbconv = atmos.turbconv
     N_up = n_updrafts(turbconv)
     gm = state
+    gm_int = state_int
     # TODO: change to new_thermo_state
     ts = recover_thermo_state(atmos, state, aux)
+    ts_int = recover_thermo_state(atmos, state_int, aux_int)
     q = PhasePartition(ts)
     _cp_m = cp_m(ts)
     lv = latent_heat_vapor(ts)
     Π = exner(ts)
     ρ_inv = 1 / gm.ρ
+    ρ_int_inv = 1 / gm.ρ
+
     surface_scalar_coeff = turbconv.surface.scalar_coeff
 
-    θ_liq_surface_flux = surf.shf / Π / _cp_m
-    q_tot_surface_flux = surf.lhf / lv
-    # these value should be given from the SurfaceFluxes.jl once it is merged
-    oblength = turbconv.surface.obukhov_length
-    ustar = turbconv.surface.ustar
+    # Retrieve surface fluxes based on MO similarity
+    ## Initial guesses for MO parameters
+    LMO_init = eps(FT)
+    u_star_init = FT(0.1)
+    th_star_init = eps(FT)
+    qt_star_init = eps(FT)
+    x_init = ArrayType(FT[LMO_init, u_star_init, th_star_init, qt_star_init])
```

Comment:
```suggestion
    x_init = MArray{Tuple{4}, FT}(LMO_init, u_star_init, th_star_init, qt_star_init)
```
We might be able to get away with using tuples for inputs: `x_init = (LMO_init, u_star_init, th_star_init, qt_star_init)` because `surface_conditions` is allocating. But `MArray`'s should work here

Reply:
Will need to apply this elsewhere